### PR TITLE
PAYARA-2545 upgrade microprofile config to 1.2.1

### PR DESCRIPTION
**Pull request by smillidge**

### CODE REVIEW

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/ClassConverter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * @author steve
+ */
+public class ClassConverter implements Converter<Class> {
+
+    @Override
+    public Class convert(String string) {
+        if (string == null || string.equals(ConfigProperty.UNCONFIGURED_VALUE)) return null;
+        try {
+            return Class.forName(string,true,Thread.currentThread().getContextClassLoader());
```
Comment (lprimak) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/ClassConverter.java]: - Use Utility.getClassLoader()/Utility.setContextClassLoader() as it works correctly w/Security Manager
Spaces between commas (string, true, etc)
Make parameter meaningful

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/CommonSenseConverter.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        if (result == null) {
+            result = convertViaValueOf(value);
+        }
+        if (result == null) {
+            result = convertViaCharSequence(value);
+        }
+        return result;
+    }
+    
+    private Object convertViaConstructor(String string) {
```
Comment (lprimak) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/CommonSenseConverter.java]: meaningful parameter / result variable name

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/CommonSenseConverter.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            // do nothing as this is normal
+        }
+        return result;
+    }
+
+    private Object convertViaCharSequence(String string) {
+        Object result = null;
+        try {
+            // need to do common sense reflected conversion
+            Method method = clazz.getMethod("parse",CharSequence.class);
```
Comment (lprimak) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/CommonSenseConverter.java]: spaces

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/CommonSenseConverter.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            result = method.invoke(null,string);
```
Comment (lprimak) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/CommonSenseConverter.java]: spaces

**Comment by smillidge** [comment]
jenkins test

**Comment by arjantijms** [comment]
jenkins test

**Pull request merged**